internal/jobs/launch_instance_aws.go

```diff
@@ -212,3 +219,47 @@ func DoLaunchInstanceAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) e
 
 	return nil
 }
+
+func FetchInstancesDescriptionAWS(ctx context.Context, args *LaunchInstanceAWSTaskArgs) error {
+	logger := ctxval.Logger(ctx)
+	logger.Debug().Msg("Started fetch instances description")
+
+	// skip job if reservation already contains errors
+	err := checkExistingError(ctx, args.ReservationID)
+	if err != nil {
+		return fmt.Errorf("step skipped: %w", err)
+	}
+
+	updateStatusBefore(ctx, args.ReservationID, "Fetching instance(s) description")
+	defer updateStatusAfter(ctx, args.ReservationID, "Instance(s) description fetched", 1)
+
+	rDao := dao.GetReservationDao(ctx)
+	instances, err := rDao.ListInstances(ctx, args.ReservationID)
+	if err != nil {
+		return fmt.Errorf("cannot get instances list: %w", err)
+	}
+	instancesIDList := make([]string, len(instances))
+	for i, instance := range instances {
+		instancesIDList[i] = instance.InstanceID
+	}
+	ec2Client, err := clients.GetEC2Client(ctx, args.ARN, args.Region)
+	if err != nil {
+		return fmt.Errorf("cannot create new ec2 client from config: %w", err)
+	}
+
+	time.Sleep(6000 * time.Millisecond)
+
+	instancesDescriptionList, err := ec2Client.ListInstancesDescription(ctx, instancesIDList)
```

Comment:
This forces the 6 second wait, shall we instead wait a second, check if ready and repeat six times?
I believe usually AWS is much faster, so there is not much reason to wait 6 seconds for all reservations, or is there some research you've done how long it takes?

Reply:
Also, there we are using the first page of the DescribeInstanceOutput, and only the [first item](https://github.com/RHEnVision/provisioning-backend/blob/67fef988a14ec93adf037e27eef77cada6d1f32d/internal/clients/http/ec2/ec2_client.go#L445) in the array.
There could be something in the NextToken field. As well as the reservation does not have to be on this index in the reservation array?
I wonder, is it safe since there is at least 6-second gap between the launching and obtaining the description?

Reply:
@avitova, as far as I understand, no need to iterate the rerservations array, because all of the instances are related to the same reservation, so only one reservation should return. 
Regarding pagination, I guess for now it is not needed, the max instances count for now is 45